### PR TITLE
File write integration. 

### PR DESCRIPTION
**Pull request by malcolmraine**
Integrated file write functions. Fixed issue with diff flagging. Moved Row class into separate file.

### CODE REVIEW

**Comment by malcolmraine** [comment]
Also added functions to convert paths in the ICONS dictionary to QIcon instances to avoid re-instantiation. Changed row highlight colors and disabled undo/redo recording while that module is being tested. Created folder for users backup files to be stored in. This will be able to be configured by the user in the future.

File: mainWindow.py [hunk 1]
```diff
@@ -37,29 +37,27 @@ def __init__(self, fileA=0, fileB=0):
...
         table_widget = main_table.MainTable(self.fIO.changesA, self.fIO.changesB)            
         #add table
         layout.addWidget(table_widget, 1, 0)
-        #table_widget.load_test_files("file1.c", "file2.c")
 
         if fileA != 0 and fileB != 0:
-            table_widget.load_table_contents([], [], fileA, fileB)    # Left list arguments for now
-        table_widget.load_table_contents([], [])    # Left list arguments for now
+            table_widget.load_table_contents(fileA, fileB)    # Left list arguments for now
+        table_widget.load_table_contents()    # Left list arguments for now
```
Comment (jmtoniolo) [inline mainWindow.py:49]: good fix, wasn’t sure of the purpose of the dummy list inputs.

File: merge_finalizer.py [hunk 2]
```diff
@@ -79,10 +81,12 @@ def backup_file(self) -> Status:
...
-        except:
+            self.backup.create_backup(f"{self.outp_file_left}", self.backup_dir)
+            self.backup.create_backup(f"{self.outp_file_right}", self.backup_dir)
+        except Exception as ex:
+            print(ex)
             return Status.BACKUP_ERROR
-        return Status.BACKUP_ERROR
+        return Status.BACKUP_SUCCESS
 
     def finalize_merge(self, left_set: list or set, right_set: list or set) -> Status:
```
Comment (jmtoniolo) [inline merge_finalizer.py:91]: Maybe a comment or two to help clarify that this is what writes to the files.

**Review by jmtoniolo — APPROVED** [review]
Looks good, couldn’t ttest on my local machine

**Pull request merged**